lib/ts/recipe/session/RequireSession.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { useContext } from "react";
+import SessionContext from "./SessionContext";
+
+type Props = {
+    requireSession?: boolean;
+};
+
+export const RequireSession: React.FC<Props> = ({ children, requireSession = true }) => {
```

Comment:
- I don't really see a point in having this refactor, especially since you can give `requireSession` as `false`.
- If this component must exist, it should not take a prop and should only show `children` if `doesSessionExist`.

Reply:
I extracted this component for higher granularity, so that `SessionAuth` can be composed out of smaller, easier testable components like this. I'm certain that such refactors will speed up future recipe development and limit false reuse.

`requireSession` is there, because it's designed to provide similar behaviour as `SessionAuth` has currently, so that it works with how we handle `requireAuth` currently.